Tasks/Xcode/xcodeutils.ts

```diff
@@ -12,6 +21,98 @@ export function getTaskState(variableName: string) {
     }
 }
 
+export function findDeveloperDir(xcodeVersion: string): string {
+    tl.debug(tl.loc('LocateXcodeBasedOnVersion', xcodeVersion));
+
+    // xcodeVersion should be in the form of "8" or "9".
+    // envName for version 9.*.* would be "XCODE_9_DEVELOPER_DIR"
+    var envName = `XCODE_${xcodeVersion}_DEVELOPER_DIR`;
+    let discoveredDeveloperDir = tl.getVariable(envName);
+    if (!discoveredDeveloperDir) {
+        discoveredDeveloperDir = fallbackDeveloperDirs[xcodeVersion];
+
+        if (discoveredDeveloperDir && !tl.exist(discoveredDeveloperDir)) {
+            tl.debug(`Ignoring fallback developer path. ${discoveredDeveloperDir} doesn't exist.`);
+            discoveredDeveloperDir = undefined;
+        }
+
+        if (!discoveredDeveloperDir) {
+            throw new Error(tl.loc('FailedToLocateSpecifiedXcode', xcodeVersion, envName));
+        }
+    }
+
+    return discoveredDeveloperDir;
+}
+
+export async function getProvisioningStyle(workspace: string) : Promise<string> {
+    var provisioningStyle: string;
+
+    if (workspace) {
+        var pbxProjectPath = getPbxProjectPath(workspace);
+        tl.debug(`pbxProjectPath is ${pbxProjectPath}`);
+
+        if (pbxProjectPath) {
+            provisioningStyle = await getProvisioningStyleFromPbxProject(pbxProjectPath);
+            tl.debug(`pbxProjectPath provisioning style: ${provisioningStyle}`);
+        }
+    }
+
+    return provisioningStyle;
+}
+
+function getPbxProjectPath(workspace: string) {
+    if (workspace && workspace.trim().toLowerCase().endsWith('.xcworkspace')) {
+        var pbxProjectPath: string = workspace.trim().toLowerCase().replace('.xcworkspace', '.pbxproj');
+
+        if (pathExistsAsFile(pbxProjectPath)) {
+            return pbxProjectPath;
+        }
+        else {
+            tl.debug("Corresponding pbxProject file doesn't exist: " + pbxProjectPath);
+        }
+    }
+}
+
+function getProvisioningStyleFromPbxProject(pbxProjectPath) : Promise<string> {
```

Comment:
Could you please add a line of documentation for the expected return types?

Reply:
Return types added to exported method.